client/src/components/EditUserProfile.jsx

```diff
@@ -0,0 +1,172 @@
+import React from 'react';
+import { Modal, Form, Button } from 'react-bootstrap';
+import Typography from '@material-ui/core/Typography';
+import Paper from '@material-ui/core/Paper';
+import { bindActionCreators } from 'redux';
+import { connect } from 'react-redux';
+import EditIcon from '@material-ui/icons/Edit';
+import '../css/EditUserProfile.css';
+import { updateUser, userLoading, uploadPreviewImage, updateProfileWithImage, getResetUserNotif } from '../actions/index';
+import { Notifier } from './Notifier.jsx';
+
+class _EditUserProfile extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      formError: null,
+      username: null,
+      email: null,
+      password: null,
+      passwordConfirm: null,
+      modalOpen: false,
+    };
+  }
+
+  _updateProfile = (event) => {
+    event.preventDefault();
+
+    const { modalOpen, formError, ...form} = { ...this.state };
+
+    // Check password and confirm password
+    if (form.password !== form.passwordConfirm) {
+      this.setState({ formError: "ERROR_SIGNUP_PASSWORD" });
+      return;
+    }
+
+    Object.keys(form).forEach(key => form[key] === null && delete form[key]);
+
+    if (Object.keys(form).length === 0) {
+      this.setState({ formError: "ERROR_EMPTY_FIELDS" });
+      return;
+    }
+
+    this.props.userLoading();
+
+    this.props.updateUser(form);
+  }
+
+  _handleChange = (evt) => {
+    this.setState({ [evt.target.name]: evt.target.value });
+  }
+
+  _openModal = () => {
+    this.setState({ modalOpen: true });
+  }
+
+  _closeModal = () => {
+    this.setState({ modalOpen: false });
+  }
+
+  _resetNotif = () => {
+    this.props.getResetUserNotif();
+  }
+
+  _handleModalChange = async (user) => {
+    const { imgFileName, previewImgFileName } = user;
+    if (imgFileName !== previewImgFileName) {
+      await this.props.updateProfileWithImage({ imgFileName: previewImgFileName });
+    }
+    this.setState({ modalOpen: false });
+  }
+
+
+  render() {
+    const { user } = this.props;
+    let notifier;
+
+    // Check form for error
+    if (this.state.formError != null) {
+      notifier = <Notifier type={this.state.formError} />
+      this.setState({ formError: null })
+    }
+
+    if (user.notifType !== null) {
+      notifier = <Notifier type={user.notifType}/>
+      this._resetNotif()
+    }
+
+    return (
+      <div className="EditProfile">
+        <Modal show={this.state.modalOpen} onHide={this._closeModal}>
+          <Modal.Header closeButton>
+            <Modal.Title>Change Profile Picture</Modal.Title>
+          </Modal.Header>
+          <Modal.Body className="update-image-body">
+            <img alt='profile img' src={user.previewImgFileName} />
```

Comment:
I would add some sizing constraints here (maybe `max-height` and `max-width` styles) so the image is scaled down to fit within the modal